### ISSUE TITLE
Chapter 09, encoder-decoder Data Preparation test_points not used for test set

### ISSUE THREAD

**Issue by gsgxnet** [dvgodoy/PyTorchStepByStep#25]
In the chunk for generation of the test set (Data Generation — Test) the `full_test`is derived from the `points` data structure, which are used for training, not from the `test_points`. 
```
test_points, test_directions = generate_sequences(seed=19)
full_test = torch.as_tensor(points).float()
source_test = full_test[:, :2]
target_test = full_test[:, 2:]
```
I do not think that is intended, so there is a simple correction possible:  
```
full_test = torch.as_tensor(test_points).float()
```
Based on that change we get different performance figures.  
Loss: 
![plot_loss](https://user-images.githubusercontent.com/8518156/151987134-312fa4f0-c198-4f21-9ad8-6696fa85e53b.png)

and another figures prediction:

![seq_pred](https://user-images.githubusercontent.com/8518156/151987406-0d0d366a-ea42-4fe5-ac17-26f2fcdc3033.png)

with 8 of 10 sequences with "clashing" points.  
If my results are right, this text chunk needs some adaption as well:  

> The results are, at the same time, very **good** and very **bad**. In ***half*** of the sequences,
the **predicted coordinates** are **quite close** to the actual ones. But, in the ***other half***,
the predicted coordinates are **overlapping with each other** and close to the
midpoint between the actual coordinates. 

> For whatever reason, the model learned
to make **good predictions** whenever the first corner is on the right edge of the
square, but really bad ones otherwise.  

See sequence pictures, these statements needs to be adapted. Especially the second.

Same issue can be found in the final putting it all together section:
```
# Validation/Test Set
test_points, test_directions = generate_sequences(seed=19)
full_test = torch.as_tensor(points).float()
source_test = full_test[:, :2]
target_test = full_test[:, 2:]
test_data = TensorDataset(source_test, target_test)
test_loader = DataLoader(test_data, batch_size=16)
```

All based on your 1.1 revision, if I did not make any mistakes in updating by `git pull`.

**Comment by gsgxnet** [comment]
The differences caused by the needed change in the `full_test`data result in significant different results also in the 
### Encoder + Decoder + PE 
section.  
I see now this loss graph:  

![Encoder + Decoder + PE loss](https://user-images.githubusercontent.com/8518156/152111989-9e325a1a-a42f-48be-a4d0-d43ef5a50502.png)

quite different from the original:  

![image](https://user-images.githubusercontent.com/8518156/152112235-6f3eea6e-234c-4b22-9144-6632ed1c34e0.png)

and in the updated test setup, we have a clashing points sequence:  

![image](https://user-images.githubusercontent.com/8518156/152112383-08b9e41a-710c-411c-b6ed-1857af1b3970.png)

please compare to the sequences based on the original code:  

![image](https://user-images.githubusercontent.com/8518156/152112515-e3044923-2892-4342-b5f1-50927540aa72.png)

**Comment by dvgodoy** [comment]
Thank you so much for pointing this out!

You're absolutely correct - it should be:
`full_test = torch.as_tensor(test_points).float()`

I will fix this and update the text to reflect the changes.

Thanks for supporting my work and helping to improve it :-)

Best,
Daniel

**Comment by dvgodoy** [comment]
Hi @gsgxnet,

I've updated code, figures, and text in the book, and published the revised edition (v1.1.1) today :-)

Once again, thanks for pointing this out.

Best,
Daniel
